File: 박승우/1463.py
n = int(input())
num = list()
num.append(0)
num.append(0)
num.append(1)
num.append(1)
for i in range(4, n + 1):
    blank = n
    blank_1 = n
    blank_2 = n
    if i % 2 == 0:
        blank = num[i // 2]
    if i % 3 == 0:
        blank_1 = num[i // 3]
    blank_2 = num[i - 1]
    num.append(min(min(blank, blank_1), blank_2) + 1)
print(num[n])

File: 박승우/1463.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int nums[1000001];

int main() {
	nums[1] = 0;
	nums[2] = 1;
	nums[3] = 1;
	int n;
	cin >> n;
	for (int i = 4; i < n + 1; i++) {
		int blank = n, blank_1 = n, blank_2 = n;
		if (i % 2 == 0)
			blank = nums[i / 2];
		if (i % 3 == 0)
			blank_1 = nums[i / 3];
		blank_2 = nums[i - 1];
		nums[i] = min(min(blank, blank_1), blank_2) + 1;
	}
	cout << nums[n] << "\n";

}
